pkg/apis/componentconfig/validation/validation.go

```diff
@@ -0,0 +1,109 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+)
+
+func ValidateKubeSchedulerConfiguration(cc *componentconfig.KubeSchedulerConfiguration) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateSchedulerAlgorithmSource(&cc.AlgorithmSource, field.NewPath("algorithmSource"))...)
+	allErrs = append(allErrs, ValidateClientConnectionConfiguration(&cc.ClientConnection, field.NewPath("clientConnection"))...)
+	allErrs = append(allErrs, ValidateKubeSchedulerLeaderElectionConfiguration(&cc.LeaderElection, field.NewPath("leaderElection"))...)
+
+	if len(cc.SchedulerName) == 0 {
+		allErrs = append(allErrs, field.Required(field.NewPath("schedulerName"), ""))
+	}
+
+	for _, msg := range validation.IsValidSocketAddr(cc.HealthzBindAddress) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("healthzBindAddress"), cc.HealthzBindAddress, msg))
+	}
+
+	for _, msg := range validation.IsValidSocketAddr(cc.MetricsBindAddress) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metricsBindAddress"), cc.MetricsBindAddress, msg))
+	}
+
+	if 0 <= cc.HardPodAffinitySymmetricWeight && cc.HardPodAffinitySymmetricWeight >= 100 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("hardPodAffinitySymmetricWeight"), cc.HardPodAffinitySymmetricWeight, "not in valid range 0-100"))
+	}
+
+	if cc.EnableContentionProfiling && !cc.EnableProfiling {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("enableContentionProfiling"), cc.EnableContentionProfiling, "can't set EnableContentionProfiling without first setting EnableProfiling"))
+	}
+
+	return allErrs
+}
+
+func ValidateClientConnectionConfiguration(cc *componentconfig.ClientConnectionConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(cc.KubeConfigFile) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("kubeConfigFile"), ""))
+	}
+	if cc.QPS < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("qps"), cc.QPS, "must be non-negative value"))
+	}
+	return allErrs
+
+}
+
+func ValidateSchedulerAlgorithmSource(cc *componentconfig.SchedulerAlgorithmSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateSchedulerPolicySource(cc.Policy, field.NewPath("schedulerPolicySource"))...)
+	return allErrs
+}
+
+func ValidateSchedulerPolicySource(cc *componentconfig.SchedulerPolicySource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cc.File == nil && cc.ConfigMap == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("file"), nil, "need to set atleast one of File or ConfigMap"))
```

Comment:
`need to set atleast one of File or ConfigMap`
->
`needs to set either File or ConfigMap`
?